File: qs.php
function qs($arr) {
  if (count($arr) < 2) {
    return $arr;
  }
  $pivot = array_shift($arr);
  $left = array_filter($arr, function($x) use ($pivot) {
    return $x <= $pivot;
  });
  $right = array_filter($arr, function($x) use ($pivot)  {
    return $x > $pivot;
  });
  return array_merge(qs($left), [$pivot], qs($right));
}

File: qs.py
def qs(vals):
  if(len(vals) < 2):
    return vals

  pivot = vals[0]
  newlist = vals[1:]
  left = filter(lambda x : x <= pivot, newlist)
  right = filter(lambda x : x > pivot, newlist)
  return qs(list(left)) + [pivot] + qs(list(right))
